kolibri/plugins/user_profile/tasks.py

```diff
@@ -159,7 +159,7 @@ def mergeuser(command, **kwargs):
 
     # check if current user should be set as superuser:
     set_as_super_user = kwargs.get("set_as_super_user")
-    if set_as_super_user:
+    if set_as_super_user and local_user.is_superuser:
```

Comment:
theoricatelly this is unneeded as this only happens if there're no more users in the device. In any case, it's harmless and prevents from an evil request from some origin different than the current kolibri frontend